src/Microsoft.AspNetCore.Authentication.Core/AuthenticationSchemeProvider.cs

```diff
@@ -19,9 +19,17 @@ public class AuthenticationSchemeProvider : IAuthenticationSchemeProvider
         /// </summary>
         /// <param name="options">The <see cref="AuthenticationOptions"/> options.</param>
         public AuthenticationSchemeProvider(IOptions<AuthenticationOptions> options)
+            : this(options, new Dictionary<string, AuthenticationScheme>(StringComparer.Ordinal))
+        {
+        }
+
+        protected AuthenticationSchemeProvider(IOptions<AuthenticationOptions> options, IDictionary<string, AuthenticationScheme> map)
```

Comment:
Add doc comments, which would be a good place to add remarks saying the map is used as is to serve scheme data.

Reply:
👍